replica/distributed_loader.cc

```diff
@@ -359,6 +359,10 @@ class table_populator {
         for (auto it = _sstable_directories.begin(); it != _sstable_directories.end(); it = _sstable_directories.erase(it)) {
             co_await it->second->stop();
         }
+
+        co_await smp::invoke_on_all([this] {
+            _global_table->mark_ready_for_writes();
+        });
```

Comment:
We have

```c++
        auto metadata = table_populator(std::move(gtable), db, ks_name, cfname);
        std::exception_ptr ex;

        try {
            co_await cf->init_storage();

            co_await metadata.start();
        } catch (...) {
            std::exception_ptr eptr = std::current_exception();
            std::string msg =
                format("Exception while populating keyspace '{}' with column family '{}' from file '{}': {}",
                        ks_name, cfname, cf->dir(), eptr);
            dblog.error("Exception while populating keyspace '{}' with column family '{}' from file '{}': {}",
                        ks_name, cfname, cf->dir(), eptr);
            try {
                std::rethrow_exception(eptr);
            } catch (sstables::compaction_stopped_exception& e) {
                // swallow compaction stopped exception, to allow clean shutdown.
            } catch (...) {
                ex = std::make_exception_ptr(std::runtime_error(msg.c_str()));
            }
        }

        co_await metadata.stop();
        if (ex) {
```

So now we will set ready_for_writes even if population fails, since stop() is called unconditionally.

Reply:
Right. It's failing the whole population anyway, but it's better to be on the safe side indeed